PWGCF/TableProducer/dptdptfilter.cxx

```diff
@@ -0,0 +1,935 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/Centrality.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/Core/TrackSelectionDefaults.h"
+#include "Common/Core/MC.h"
+#include "Common/Core/PID/PIDResponse.h"
+#include "PWGCF/Core/AnalysisConfigurableCuts.h"
+#include "PWGCF/DataModel/DptDptFiltered.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Framework/runDataProcessing.h"
+#include <TROOT.h>
+#include <TDatabasePDG.h>
+#include <TParameter.h>
+#include <TList.h>
+#include <TDirectory.h>
+#include <TFolder.h>
+#include <TH1.h>
+#include <TH2.h>
+#include <TH3.h>
+#include <TProfile3D.h>
+
+#include <cmath>
+
+#include "dptdptfilter.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::soa;
+using namespace o2::framework::expressions;
+using namespace o2::analysis;
+
+#define DPTDPTFILTERLOGCOLLISIONS debug
+#define DPTDPTFILTERLOGTRACKS debug
+
+namespace o2::analysis::dptdptfilter
+{
+using DptDptFullTracksPID = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksExtended, aod::TrackSelection, aod::pidTPCEl, aod::pidTPCMu, aod::pidTPCPi, aod::pidTPCKa, aod::pidTPCPr, aod::pidTOFEl, aod::pidTOFMu, aod::pidTOFPi, aod::pidTOFKa, aod::pidTOFPr>;
+using DptDptFullTracksPIDDetLevel = soa::Join<aod::Tracks, aod::McTrackLabels, aod::TracksExtra, aod::TracksExtended, aod::TrackSelection, aod::pidTPCEl, aod::pidTPCMu, aod::pidTPCPi, aod::pidTPCKa, aod::pidTPCPr, aod::pidTOFEl, aod::pidTOFMu, aod::pidTOFPi, aod::pidTOFKa, aod::pidTOFPr>;
+using DptDptFullTracks = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksExtended, aod::TrackSelection>;
+using DptDptFullTracksDetLevel = soa::Join<aod::Tracks, aod::McTrackLabels, aod::TracksExtra, aod::TracksExtended, aod::TrackSelection>;
+
+/// \enum MatchRecoGenSpecies
+/// \brief The species considered by the matching tast
+enum MatchRecoGenSpecies {
+  kDptDptCharged = 0, ///< charged particle/track
+  kDptDptElectron,    ///< electron
+  kDptDptMuon,        ///< muon
+  kDptDptPion,        ///< pion
+  kDptDptKaon,        ///< kaon
+  kDptDptProton,      ///< proton
+  kDptDptNoOfSpecies, ///< the number of considered species
+  kWrongSpecies = -1
+};
+
+const char* speciesName[kDptDptNoOfSpecies] = {"h", "e", "mu", "pi", "ka", "p"};
+
+const char* speciesTitle[kDptDptNoOfSpecies] = {"", "e", "#mu", "#pi", "K", "p"};
+
+//============================================================================================
+// The DptDptFilter output objects
+//============================================================================================
+std::string fTaskConfigurationString = "PendingToConfigure";
+TH1F* fhCentMultB = nullptr;
```

Comment:
No histogram registry?

Reply:
Not yet for the filter task
It has been introduced in the match reco-gen and in the filter-qa